.github/workflows/harbor-testnet.yml

```diff
@@ -0,0 +1,138 @@
+          aws-region: ${{ secrets.HARBOR_REGION }}
+
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_REGISTRATION_TOKEN }}
+          ec2-image-id: ami-012b908cf83070015
+          ec2-instance-type: c5.xlarge
+          subnet-id: subnet-0ff82089478a04bd1
+          security-group-id: sg-0a2341bf0d00efdee
+
+  bring-testnet-up:
+    name: Build the testnet on the runner
+    needs: init # required to start the main job when the runner is ready
+    runs-on: ${{ needs.init.outputs.label }} # run the job on the newly created runner
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GH_REGISTRATION_TOKEN }}
+          ref: ${{ github.event.workflow_run.head_sha }}
+
+      - name: Setup Nodejs and npm
+        uses: actions/setup-node@v2
+        with:
+          node-version: "18"
+
+      - name: Configure Harbor
+        run: /root/go/bin/harbor configure keys --user-key ${{ secrets.HARBOR_USER_KEY }} --project-key ${{ secrets.HARBOR_PROJECT_KEY }}
+
+      - name: Compile code and other dependencies
+        run: yarn workspaces focus @connext/nxtp-contracts && yarn workspace @connext/nxtp-contracts build
+
+      - name: Build and Run Testnet
+        run: /root/go/bin/harbor apply ${{ needs.init.outputs.short-sha }} --placeholder-map sha_default=${{ needs.init.outputs.short-sha }},tag_default=${{ needs.init.outputs.short-sha }} --no-wait true 
+
+      - name: yarn && yarn build:all
+        run: |
+          export NODE_OPTIONS="--max-old-space-size=6144" 
+          yarn && yarn build:all
+
+      - name: Poll until testnet running
+        run: sh /root/describe.sh ${{ needs.init.outputs.short-sha }}
+
+      - name: Describe testnet
+        id: testnet-data
+        run: |
+          TDATA=$( /root/go/bin/harbor describe testnet --name ${{ needs.init.outputs.short-sha }} --output json | sed '1d' )
+          TDATA="${TDATA//'%'/''}"
+          TDATA="${TDATA//$'\n'/''}"
+          TDATA="${TDATA//$'\r'/''}"
+          echo "TDATA=$TDATA" >> $GITHUB_OUTPUT
+          
+      - name: Echo testnet details 
+        run: echo ${{ steps.testnet-data.outputs.TDATA }}
+
+      - name: Echo testnet name
+        run: |
+          echo "${{fromJson(steps.testnet-data.outputs.TDATA).name}}"
+      
+      - name: Build and deploy subgraph 
+        run: |
+          IPFS_URL="${{fromJson(steps.testnet-data.outputs.TDATA).offChainActors.ipfs.endpoints[0]}}"
+          GRAPH_URL_1337="${{fromJson(steps.testnet-data.outputs.TDATA).offChainActors.graphNode1337.endpoints[0]}}" 
+          GRAPH_URL_1338="${{fromJson(steps.testnet-data.outputs.TDATA).offChainActors.graphNode1338.endpoints[0]}}"
+          IPFS_URL="${IPFS_URL%%:*}" GRAPH_URL_1337="${GRAPH_URL_1337%%:*}" GRAPH_URL_1338="${GRAPH_URL_1338%%:*}" sh setup_subgraph.sh
```

Comment:
What would run the rest of the setup currently done in:
`setup-integration-test.sh`

Reply:
So everything in setup-integration-test.sh is now removable - just run the harbor apply  command and everything will get configured. In the coming 2 weeks we're shipping a SDK so that this is even easier .

Reply:
Just a note - in the version we're running here we autopopulate the tags and bump them. This isn't in the latest version of the CLI yet, so locally you'll have to change the tags.

We expect this merge to go in in the next couple of days so you guys don't have to worry about the tags at all.

Reply:
@preethamr Here, we daisy-chain our action with the E2E Workflow where the build happens. We use the images built on the E2E Workflow while setting up our harbor testnet.